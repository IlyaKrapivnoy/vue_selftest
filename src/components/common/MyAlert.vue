<template>
  <div v-show="props.isAlert" :class="props.wrapperClass">
    <el-alert
      :title="props.title"
      :type="props.type"
      :description="props.description"
      :show-icon="props.showIcon"
      :closable="props.closable"
    />
  </div>
</template>

<script setup>
const props = defineProps({
  isAlert: Boolean,
  wrapperClass: String,
  title: String,
  type: {
    validator: (value) =>
      ["success", "warning", "error", "info"].includes(value),
    default: "success",
  },
  description: String,
  showIcon: Boolean,
  closable: Boolean,
});
</script>
