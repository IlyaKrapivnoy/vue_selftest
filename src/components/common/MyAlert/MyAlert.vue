<template>
  <div v-show="props.isAlert" :class="props.wrapperClass">
    <el-alert
      :title="props.title ?? ''"
      :type="props.type"
      :description="props.description ?? ''"
      :show-icon="props.showIcon"
      :closable="props.closable"
    />
  </div>
</template>

<script setup>
const props = defineProps({
  isAlert: {
    type: Boolean,
    default: false,
  },
  wrapperClass: {
    type: String,
    default: "",
  },
  title: {
    type: String,
    default: "",
  },
  type: {
    type: String,
    validator: (value) =>
      ["success", "warning", "error", "info"].includes(value),
    default: "success",
  },
  description: {
    type: String,
    default: "",
  },
  showIcon: {
    type: Boolean,
    default: true,
  },
  closable: {
    type: Boolean,
    default: true,
  },
});
</script>
