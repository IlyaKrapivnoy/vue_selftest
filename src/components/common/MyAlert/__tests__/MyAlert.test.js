import { mount } from "@vue/test-utils";
import MyAlert from "@/components/common/MyAlert/MyAlert.vue";

describe("MyAlert", () => {
  it("renders correctly when isAlert is true", () => {
    const wrapperClass = "my-alert-wrapper";
    const title = "Alert Title";
    const type = "warning";
    const description = "This is the alert description";
    const showIcon = true;
    const closable = true;

    const wrapper = mount(MyAlert, {
      props: {
        isAlert: true,
        wrapperClass,
        title,
        type,
        description,
        showIcon,
        closable,
      },
    });

    expect(wrapper.exists()).toBe(true);
    expect(wrapper.classes()).toContain(wrapperClass);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
