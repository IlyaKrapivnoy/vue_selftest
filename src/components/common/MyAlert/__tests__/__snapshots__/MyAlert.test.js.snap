// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAlert renders correctly when isAlert is true 1`] = `
"<div class="my-alert-wrapper">
  <el-alert title="Alert Title" type="warning" description="This is the alert description" show-icon="true" closable="true"></el-alert>
</div>"
`;
